### PR TITLE
US3463 Construct MRCPPrompt for :native_or_unimrcp recognizer

### PR DESCRIPTION
**Pull request by Jared-Prime**
Add ':native_or_unimrcp' as an input recognizer that will construct an `MRCPPrompt` rather than a `ComposedPrompt`. This is a precursor to [enabling mixed audio and TTS](https://github.com/cloudvox/apptastic/issues/144) to run with ASR enabled.

### CODE REVIEW

**Comment by Jared-Prime** [comment]
Only tangentially related to [this PR being closed](https://github.com/adhearsion/punchblock/issues/222), I think may be going about this incorrectly.

**Comment by Jared-Prime** [comment]
@runninferret @sfgeorge please review

File: lib/punchblock/translator/asterisk/component/mrcp_prompt.rb [hunk 1]
```diff
@@ -11,11 +11,11 @@ class MRCPPrompt < Component
...
-            raise OptionError, "The renderer #{renderer} is unsupported." unless renderer == 'unimrcp'
-            raise OptionError, "The recognizer #{recognizer} is unsupported." unless recognizer == 'unimrcp'
-            raise OptionError, 'An SSML document is required.' unless output_node.render_documents.count > 0
-            raise OptionError, 'Only one document is allowed.' if output_node.render_documents.count > 1
-            raise OptionError, 'A grammar is required.' unless input_node.grammars.count > 0
+            fail OptionError, "The renderer #{renderer} is unsupported." unless renderer == 'unimrcp' || renderer == 'native_or_unimrcp'
+            fail OptionError, "The recognizer #{recognizer} is unsupported." unless recognizer == 'unimrcp' || renderer == 'native_or_unimrcp'
+            fail OptionError, 'An SSML document is required.' unless output_node.render_documents.count > 0
+            fail OptionError, 'Only one document is allowed.' if output_node.render_documents.count > 1
+            fail OptionError, 'A grammar is required.' unless input_node.grammars.count > 0
```
Comment (sfgeorge) [inline lib/punchblock/translator/asterisk/component/mrcp_prompt.rb]: In call.rb and mrcp_prompt.rb, I would refrain from making code style changes to Punchblock.  Good idea in spirit, but makes only increases the diff between our fork and the mainline, making it more difficult to bridge the gap.

**Comment by Jared-Prime** [comment]
@sfgeorge reverted

**Comment by sfgeorge** [comment]
Thanks man.   Can I be a wet blanket and also ask you to eliminate the nice and clean `#determine_component_class` method?  Sorry :/

**Comment by sfgeorge** [comment]
Can you add an entry to the ifbyphone [CHANGELOG](https://github.com/cloudvox/punchblock/blob/develop-ifbyphone/CHANGELOG.md) with a note about this change?  On a related note, you may need to rebase your PR 1st in order to see our latest CHANGELOG entries and keep this PR mergeable.

**Comment by Jared-Prime** [comment]
rebased and changelog updated

**Comment by sfgeorge** [comment]
Thanks for all the changes man.  This looks great. :+1: 

@ajayanty Can you merge this for us?

**Comment by ajayanty** [comment]
This has been merged into  cloudvox:develop-ifbyphone

**Pull request merged**